### PR TITLE
Remove leftover room code

### LINKED ISSUE #6824: Remove vestigial room code from transport
## Abstract

With #6753 the usage of rooms for node-to-node communication was removed.
A few pieces of room handling code remain for now to catch possible implementation bugs. These are:
- The `Room` member in `MatrixRoomMessages` and it's use in `_handle_sync_messages()`
- The `handle_member_join()` method

These should be removed after some time, e.g. after the next mainnet release.

## Motivation

Less code is good.

## Specification

Remove it

## Backwards Compatibility

N/A

### PR DESCRIPTION
**Pull request by istankovic**
Fixes: #6824

### CODE REVIEW

File: raiden/network/transport/matrix/client.py [hunk 1]
```diff
@@ -833,15 +825,10 @@ def _handle_responses(self, currently_queued_responses: List[JSONResponse]) -> N
...
-                        [
-                            message
-                            for message in sync_room["timeline"]["events"]
-                            if message["type"] == "m.room.message"
-                        ],
-                    )
+                all_messages.extend(
+                    message
+                    for message in sync_room["timeline"]["events"]
+                    if message["type"] == "m.room.message"
```
Comment (fredo) [inline raiden/network/transport/matrix/client.py]: actually there should not be any messages arriving through rooms anymore. I have no strong feelings if we should leave adding the messages to `all_messages`

**Review by fredo — APPROVED** [review]
LGTM although I haven't checked the untouched code parts if there are still room leftovers.

File: raiden/network/transport/matrix/client.py (hunk 1 above)
Comment (istankovic) [reply raiden/network/transport/matrix/client.py]: Hm, good point, I'll remove it.

**Review by fredo — APPROVED** [review]
looks good. At one point we should revisit the sync filter which is probably not necessary anymore. BNut this can wait for another PR.

**Pull request merged**